### PR TITLE
upump: add restart control for timers

### PR DESCRIPTION
**Pull request by quarium**
Add upump_restart for timer pumps to reset the timeout.

### CODE REVIEW

**Comment by cmassiot** [comment]
It should be explained somewhere what it does and what the difference with upump_start is.

File: include/upipe/upump.h [hunk 1]
```diff
@@ -346,7 +346,10 @@ static inline void upump_stop(struct upump *upump)
     upump_control(upump, UPUMP_STOP);
 }
 
-/** @This asks the event loop to restart a monitoring pump.
+/** @This asks the event loop to restart a timer pump.
+ * The timer will be restart to its initial value, as if it was stop, set to
+ * its inital value and start. This could be usefull to implement IO timeout
```
Comment (nto) [inline include/upipe/upump.h]: "useful"

**Comment by cmassiot** [comment]
can you squash it ?

**Comment by quarium** [comment]
Squashed

**Pull request merged**